akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEvent.scala

```diff
@@ -113,23 +133,39 @@ object ServerSentEvent {
  * @param id event id, must not contain \n or \r
  * @param retry the reconnection time in milliseconds.
  */
-final case class ServerSentEvent(data: String, eventType: Option[String] = None, id: Option[String] = None, retry: Option[Int] = None) {
+final case class ServerSentEvent(
+  data: String,
+  eventType: Option[String] = None,
+  id: Option[String] = None,
+  retry: Option[Int] = None
+)
+    extends EventStreamElement {
   import ServerSentEvent._
   require(eventType.forall(_.forall(c => c != '\n' && c != '\r')), "Event type must not contain \\n or \\r!")
   require(id.forall(_.forall(c => c != '\n' && c != '\r')), "Id must not contain \\n or \\r!")
   require(retry.forall(_ > 0L), "Retry must be a positive number!")
 
   /**
+   * Encodes this Server-Sent Event to an `akka.util.ByteString` according to the
+   * [[http://www.w3.org/TR/eventsource/#event-stream-interpretation SSE specification]].
+   *
+   * @return message converted to UTF-8 encoded `akka.util.ByteString`
+   */
+  override def encode = ByteString(toString, UTF_8.name)
```

Comment:
@hseeberger But it is done in one chunk no matter how much of it is actually possible to transfer at that point in time. By constructing it in smaller chunks there will be less resident memory usage under backpressure.